### PR TITLE
Do not recreate viz table in collection_spec

### PR DESCRIPTION
**Pull request by javitonino**
`collection_spec` creates copies of the visualization tables to perform its tests. However, some tests (namely, testing the visualizations locator) use the main table, but still recreate it every time.

This PR avoids this situation by never recreating the main table and ensuring that the visualization repository is left pointing to the main table (instead of one of the temp tables). This fixes Maps API controller tests in master CI.

It also updates the Migrator schema to the current schema so tests are run against the real table.

See #6848

### CODE REVIEW

**Comment by javitonino** [comment]
:+1:

**Comment by javitonino** [comment]
CR @gfiorav

File: spec/models/visualization/collection_spec.rb [hunk 1]
```diff
@@ -47,6 +40,10 @@
     @user_2.destroy
   end
 
+  def restore_vis_backend_to_normal_table_so_relator_works
```
Comment (gfiorav) [inline spec/models/visualization/collection_spec.rb]: I know you inherited this, but maybe it might be a good time to rename this method!

**Comment by gfiorav** [comment]
Except for that single comment about the method name, :package: it!

Nice work :)

**Pull request merged**